Explorer/Assets/DCL/Settings/ModuleControllers/GraphicsQualitySettingsController.cs

```diff
@@ -10,33 +14,96 @@ public class GraphicsQualitySettingsController : SettingsFeatureController
         private const string GRAPHICS_QUALITY_DATA_STORE_KEY = "Settings_GraphicsQuality";
 
         private readonly SettingsDropdownModuleView view;
+        private readonly RealmPartitionSettingsAsset realmPartitionSettingsAsset;
+        private readonly LandscapeData landscapeData;
+        private readonly QualitySettingsAsset qualitySettingsAsset;
 
-        public GraphicsQualitySettingsController(SettingsDropdownModuleView view)
+        public GraphicsQualitySettingsController(SettingsDropdownModuleView view, RealmPartitionSettingsAsset realmPartitionSettingsAsset, LandscapeData landscapeData, QualitySettingsAsset qualitySettingsAsset)
         {
             this.view = view;
 
-            // Clean current options loaded from the settings menu configuration and load names from QualitySettings
-            view.DropdownView.Dropdown.options.Clear();
-            foreach (string option in QualitySettings.names)
-                view.DropdownView.Dropdown.options.Add(new TMP_Dropdown.OptionData { text = option });
+            this.realmPartitionSettingsAsset = realmPartitionSettingsAsset;
+            this.landscapeData = landscapeData;
+            this.qualitySettingsAsset = qualitySettingsAsset;
+
+            LoadGraphicsQualityOptions();
 
             view.DropdownView.Dropdown.value = settingsDataStore.HasKey(GRAPHICS_QUALITY_DATA_STORE_KEY) ?
                 settingsDataStore.GetDropdownValue(GRAPHICS_QUALITY_DATA_STORE_KEY) :
                 DEFAULT_QUALITY_LEVEL_INDEX;
 
             view.DropdownView.Dropdown.onValueChanged.AddListener(SetGraphicsQualitySettings);
             SetGraphicsQualitySettings(view.DropdownView.Dropdown.value);
+
+            realmPartitionSettingsAsset.OnMaxLoadingDistanceInParcelsChanged += OnMaxLoadingDistanceInParcelsChanged;
+            landscapeData.OnDetailDistanceChanged += OnDetailDistanceChanged;
+        }
+
+        private void LoadGraphicsQualityOptions()
+        {
+            view.DropdownView.Dropdown.options.Clear();
+            foreach (string option in QualitySettings.names)
+                view.DropdownView.Dropdown.options.Add(new TMP_Dropdown.OptionData { text = option });
+            view.DropdownView.Dropdown.options.Add(new TMP_Dropdown.OptionData { text = "Custom" });
         }
 
         private void SetGraphicsQualitySettings(int index)
         {
-            QualitySettings.SetQualityLevel(index);
+            if (index < view.DropdownView.Dropdown.options.Count - 1)
+                ForceSetQualityLevel(index);
+
             settingsDataStore.SetDropdownValue(GRAPHICS_QUALITY_DATA_STORE_KEY, index, save: true);
         }
 
+        private void ForceSetQualityLevel(int index)
+        {
+            int currentQualityLevel = QualitySettings.GetQualityLevel();
```

Comment:
I am a bit confused at this logic, if we retrieve the current quality level and its the same why do we need to do any logic?